packages/jest-matchers/src/index.js

```diff
@@ -57,6 +57,7 @@ if (!global[GLOBAL_STATE]) {
       state: {
         assertionCalls: 0,
         assertionsExpected: null,
+        expectedAssertions: false,
```

Comment:
How about:
```json
expectedAssertionsNumber: null, 
isExpectingAssertions: false,
```
Seems a bit more descriptive. `assertionsExpected` and `expectedAssertions` don't tell much to me.